pkg/services/navtree/navtreeimpl/navtree.go

```diff
@@ -345,22 +345,17 @@ func (s *ServiceImpl) buildStarredItemsNavLinks(c *contextmodel.ReqContext) ([]*
 		return nil, err
 	}
 
-	starredDashboards := []*dashboards.Dashboard{}
-	starredDashboardsCounter := 0
-	for dashboardId := range starredDashboardResult.UserStars {
-		// Set a loose limit to the first 50 starred dashboards found
-		if starredDashboardsCounter > 50 {
-			break
-		}
-		starredDashboardsCounter++
-		query := &dashboards.GetDashboardQuery{
-			ID:    dashboardId,
-			OrgID: c.OrgID,
-		}
-		queryResult, err := s.dashboardService.GetDashboard(c.Req.Context(), query)
-		if err == nil {
-			starredDashboards = append(starredDashboards, queryResult)
-		}
+	var ids []int64
+	for id := range starredDashboardResult.UserStars {
+		ids = append(ids, id)
+	}
+	starredDashboards, err := s.dashboardService.GetDashboards(c.Req.Context(), &dashboards.GetDashboardsQuery{DashboardIDs: ids, OrgID: c.OrgID})
+	if err != nil {
+		return nil, err
+	}
+	// Set a loose limit to the first 50 starred dashboards found
+	if len(starredDashboards) > 50 {
+		starredDashboards = starredDashboards[:50]
```

Comment:
Starred dashboard table doesn't store org ID for the org that the dashboard is in, so it could be the case that the first 50 starred dashboard IDs are from a different org. Therefore I think we need to query for all of them unfortunately.

Reply:
I see

Reply:
It's been a long day so I might not be looking at this right. These `starredDashboards` are getting fetched with this [GetDashboards](https://github.com/grafana/grafana/blob/f013d37ce965bd5789856113c2e36310da1c45e7/pkg/services/navtree/navtreeimpl/navtree.go#L353) call which wraps around [FindDashboards](https://github.com/grafana/grafana/blob/main/pkg/services/dashboards/database/database.go#L915). That's not the query to starred dashboard table that you're talking about is it?